vscode-dotnet-runtime-library/src/Acquisition/LinuxVersionResolver.ts

```diff
@@ -0,0 +1,288 @@
+            : await this.distroSDKProvider!.getExpectedDotnetMicrosoftFeedInstallationDirectory() ))
+        {
+            const err = new DotnetCustomLinuxInstallExistsError(new Error(this.conflictingCustomInstallErrorMessage + existingInstall),
+            fullySpecifiedDotnetVersion);
+            this.acquisitionContext.eventStream.post(err);
+            throw err;
+        }
+    }
+
+    /**
+     *
+     * @param fullySpecifiedDotnetVersion The version to install of the dotnet sdk.
+     * @param existingInstall a path to the existing dotnet install on the machine.
+     * @returns 0 if we can proceed. Will throw if a conflicting install exists. If we can update, it will do the update and return 1.
+     * A string is returned in case we want to make this return more info about the update.
+     */
+    private async UpdateOrRejectIfVersionRequestDoesNotRequireInstall(fullySpecifiedDotnetVersion : string, existingInstall : string | null)
+    {
+        await this.Initialize();
+
+        if(existingInstall)
+        {
+            const existingGlobalInstallSDKVersion = await this.distroSDKProvider!.getInstalledGlobalDotnetVersionIfExists();
+            if(existingGlobalInstallSDKVersion && Number(VersionResolver.getMajorMinor(existingGlobalInstallSDKVersion)) === Number(VersionResolver.getMajorMinor(fullySpecifiedDotnetVersion)))
+            {
+                if(Number(VersionResolver.getMajorMinor(existingGlobalInstallSDKVersion)) > Number(VersionResolver.getMajorMinor(fullySpecifiedDotnetVersion)))
+                {
+                    // We shouldn't downgrade to a lower patch
+                    const err = new DotnetCustomLinuxInstallExistsError(new Error(`An installation of ${fullySpecifiedDotnetVersion} was requested but ${existingGlobalInstallSDKVersion} is already available.`),
+                        fullySpecifiedDotnetVersion);
+                    this.acquisitionContext.eventStream.post(err);
+                    throw err;
+                }
+                else if(await this.distroSDKProvider!.dotnetPackageExistsOnSystem(fullySpecifiedDotnetVersion) ||
+                    Number(VersionResolver.getFeatureBandPatchVersion(existingGlobalInstallSDKVersion)) < Number(VersionResolver.getFeatureBandPatchVersion(fullySpecifiedDotnetVersion)))
+                {
+                    // We can update instead of doing an install
+                    return (await this.distroSDKProvider!.upgradeDotnet(existingGlobalInstallSDKVersion)) ? '1' : '1';
+                }
+                else
+                {
+                    // An existing install exists.
+                    return '1';
+                }
+            }
+            // Additional logic to check the major.minor could be added here if we wanted to prevent installing lower major.minors if an existing install existed.
+        }
+        return '0';
```

Comment:
I went ahead and changed this, but then discovered there is no int type, it is only number.... I would rather return the string than a potential float exit code since it gives us the ability to return other error information if needed. I agree that it's not super ideal though